packages/block-editor/src/store/selectors.js

```diff
@@ -124,13 +124,21 @@ export function getBlockAttributes( state, clientId ) {
  * the template block itself is considered part of the parent, but the children
  * are not.
  *
- * @param {Object} state    Editor state.
- * @param {string} clientId Block client ID.
+ * You can override this behavior with the includeControlledInnerBlocks. So if
+ * you call `getBlock( TP, true )`, it will return all nested blocks, including
+ * all child inner block controllers and their children.
+ *
+ * @param {Object}   state                        Editor state.
+ * @param {string}   clientId                     Block client ID.
+ * @param {?boolean} includeControlledInnerBlocks If true, include controlleld
+ *                                                inner block subtrees. The default
+ *                                                behavior is to exclude controlled
+ *                                                inner blocks (false).
  *
  * @return {Object} Parsed block object.
  */
 export const getBlock = createSelector(
-	( state, clientId ) => {
+	( state, clientId, includeControlledInnerBlocks = false ) => {
```

Comment:
Personally, I prefer when functions put their behavior option arguments in an object, e.g. `state, clientId, { includeControlledInnerBlocks = false }`, since it makes adding (or deprecating) options easier without having to pass `undefined`/`null` to other arguments that were added to the function at an earlier point in development. It also makes the code more clear, compared to just passing `true` as the third argument.

I'm not sure if this function is one that is likely to ever have another optional parameter added to it, but I think it's good practice to design functions like that anyway.

But that's just my opinion. What do other people think?

Reply:
That makes sense to me :)